### PR TITLE
New package: flameshot

### PR DESCRIPTION
**Pull request by ArchGuyWu**

### CODE REVIEW

**Comment by ArchGuyWu** [comment]
Flameshot seems not officially support MinGW, and needs a small patch to make it build using MinGW, should I continue to package it to msys2?

**Comment by ognevny** [comment]
we have a lot of software that require patches to make it compileable/workable there. so it just depends on your motivation

**Comment by ArchGuyWu** [comment]
> we have a lot of software that require patches to make it compileable/workable there. so it just depends on your motivation

Ok, it's just a small patch.

**Comment by ognevny** [comment]
feel free to push it!

File: mingw-w64-flameshot/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+prepare() {
+  cd "${srcdir}/${_realname}"
+
+  # Hardcode FLAMESHOT_VERSION to avoid syntax error
+  sed -i "/VALUE \"FileVersion\"/s/VER_FILEVERSION_STR/\"v$pkgver\"/" data/flameshot.rc
+  sed -i "/VALUE \"ProductVersion\"/s/VER_PRODUCTVERSION_STR/\"v$pkgver\"/" data/flameshot.rc
+
+  # Remove all QCP_EXPORT used by MSVC
+  find "external/Qt-Color-Widgets/include/QtColorWidgets" -type f -name "*.hpp" \
+    -exec sed -i 's/class QCP_EXPORT/class/g' {} +
```
[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:31]: The affected files are up to 20, and don't know whether the number will increase in the future.
https://github.com/search?q=repo%3Aflameshot-org%2Fflameshot%20class%20QCP_EXPORT&type=code

[the remark this answers is not in the document]
Comment (ognevny) [reply mingw-w64-flameshot/PKGBUILD:31]: got it. so I'll wait for others to review it (I have a bad experience with such things)

[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:31]: Ok, and I found that missing some files compared to Arch Linux.

[the remark this answers is not in the document]
Comment (ognevny) [reply mingw-w64-flameshot/PKGBUILD:31]: which files? I guess that the files that required for Linux distros only

[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:31]: translations and maybe icons

[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:31]: > > maybe icons
> 
> separate files for icons are required for Linux only as in Windows they are part of .exe file (.ico files is compiled with manifest and linked into .exe)

Ok, so where should we place those translations?

[the remark this answers is not in the document]
Comment (ognevny) [reply mingw-w64-flameshot/PKGBUILD:31]: maybe `${MINGW_PREFIX}/share/flameshot/translations/`?

[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:31]: The translations path defines here: https://github.com/flameshot-org/flameshot/blob/fd3772e2abb0b852573fcaa549ba13517f13555c/src/utils/pathinfo.cpp#L23 , we can directly patch it.

[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:31]: > maybe `${MINGW_PREFIX}/share/flameshot/translations/`? if it's not installed by build system than probably it's not supported in Windows, but it worths to try

I installed it by scoop, and saw that translations placed in bin directory.

**Comment by ArchGuyWu** [comment]
Unable to locate compiled translations.

File: mingw-w64-flameshot/PKGBUILD (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArchGuyWu) [reply mingw-w64-flameshot/PKGBUILD:33]: The translation files did compiled, but I'm unable to locate them.

**Comment by ArchGuyWu** [comment]
Can you explain this? @ognevny 

```
user@WIN-SERVER CLANG64 /c/Users/user/Downloads/build
$ ls src/build-CLANG64/src/Internationalization_*.qm
src/build-CLANG64/src/Internationalization_bg.qm     src/build-CLANG64/src/Internationalization_fr.qm     src/build-CLANG64/src/Internationalization_ko.qm     src/build-CLANG64/src/Internationalization_sv_SE.qm
src/build-CLANG64/src/Internationalization_ca.qm     src/build-CLANG64/src/Internationalization_gl.qm     src/build-CLANG64/src/Internationalization_nb_NO.qm  src/build-CLANG64/src/Internationalization_tr.qm
src/build-CLANG64/src/Internationalization_cs.qm     src/build-CLANG64/src/Internationalization_grc.qm    src/build-CLANG64/src/Internationalization_nl.qm     src/build-CLANG64/src/Internationalization_uk.qm
src/build-CLANG64/src/Internationalization_de_DE.qm  src/build-CLANG64/src/Internationalization_he.qm     src/build-CLANG64/src/Internationalization_nl_NL.qm  src/build-CLANG64/src/Internationalization_vi.qm
src/build-CLANG64/src/Internationalization_el.qm     src/build-CLANG64/src/Internationalization_hu.qm     src/build-CLANG64/src/Internationalization_pl.qm     src/build-CLANG64/src/Internationalization_zh_CN.qm
src/build-CLANG64/src/Internationalization_es.qm     src/build-CLANG64/src/Internationalization_id.qm     src/build-CLANG64/src/Internationalization_pt_BR.qm  src/build-CLANG64/src/Internationalization_zh_HK.qm
src/build-CLANG64/src/Internationalization_eu.qm     src/build-CLANG64/src/Internationalization_it_IT.qm  src/build-CLANG64/src/Internationalization_ru.qm     src/build-CLANG64/src/Internationalization_zh_TW.qm
src/build-CLANG64/src/Internationalization_fa.qm     src/build-CLANG64/src/Internationalization_ja.qm     src/build-CLANG64/src/Internationalization_sk.qm
src/build-CLANG64/src/Internationalization_fi.qm     src/build-CLANG64/src/Internationalization_ka.qm     src/build-CLANG64/src/Internationalization_sr_SP.qm

user@WIN-SERVER CLANG64 /c/Users/user/Downloads/build
$ ls src/build-CLANG64/src/lib*.dll.a
src/build-CLANG64/src/libflameshot.dll.a
```

https://github.com/msys2/MINGW-packages/actions/runs/10353019056

**Comment by ArchGuyWu** [comment]
> Can you explain this? @lazka
> 
> ```
> user@WIN-SERVER CLANG64 /c/Users/user/Downloads/build
> $ ls src/build-CLANG64/src/Internationalization_*.qm
> src/build-CLANG64/src/Internationalization_bg.qm     src/build-CLANG64/src/Internationalization_fr.qm     src/build-CLANG64/src/Internationalization_ko.qm     src/build-CLANG64/src/Internationalization_sv_SE.qm
> src/build-CLANG64/src/Internationalization_ca.qm     src/build-CLANG64/src/Internationalization_gl.qm     src/build-CLANG64/src/Internationalization_nb_NO.qm  src/build-CLANG64/src/Internationalization_tr.qm
> src/build-CLANG64/src/Internationalization_cs.qm     src/build-CLANG64/src/Internationalization_grc.qm    src/build-CLANG64/src/Internationalization_nl.qm     src/build-CLANG64/src/Internationalization_uk.qm
> src/build-CLANG64/src/Internationalization_de_DE.qm  src/build-CLANG64/src/Internationalization_he.qm     src/build-CLANG64/src/Internationalization_nl_NL.qm  src/build-CLANG64/src/Internationalization_vi.qm
> src/build-CLANG64/src/Internationalization_el.qm     src/build-CLANG64/src/Internationalization_hu.qm     src/build-CLANG64/src/Internationalization_pl.qm     src/build-CLANG64/src/Internationalization_zh_CN.qm
> src/build-CLANG64/src/Internationalization_es.qm     src/build-CLANG64/src/Internationalization_id.qm     src/build-CLANG64/src/Internationalization_pt_BR.qm  src/build-CLANG64/src/Internationalization_zh_HK.qm
> src/build-CLANG64/src/Internationalization_eu.qm     src/build-CLANG64/src/Internationalization_it_IT.qm  src/build-CLANG64/src/Internationalization_ru.qm     src/build-CLANG64/src/Internationalization_zh_TW.qm
> src/build-CLANG64/src/Internationalization_fa.qm     src/build-CLANG64/src/Internationalization_ja.qm     src/build-CLANG64/src/Internationalization_sk.qm
> src/build-CLANG64/src/Internationalization_fi.qm     src/build-CLANG64/src/Internationalization_ka.qm     src/build-CLANG64/src/Internationalization_sr_SP.qm
> 
> user@WIN-SERVER CLANG64 /c/Users/user/Downloads/build
> $ ls src/build-CLANG64/src/lib*.dll.a
> src/build-CLANG64/src/libflameshot.dll.a
> ```
> 
> https://github.com/msys2/MINGW-packages/actions/runs/10353019056

It's ready to merge now, I changed the translations packaging script.

**Review by ognevny — APPROVED** [review]
will leave an approve but I want someone to review `QCP_EXPORT` removal

**Comment by Biswa96** [comment]
> will leave an approve but I want someone to review `QCP_EXPORT` removal

Why was it removed?

**Comment by ArchGuyWu** [comment]
> > will leave an approve but I want someone to review `QCP_EXPORT` removal
> 
> Why was it removed?

https://github.com/flameshot-org/flameshot/issues/2524#issuecomment-1756654273

**Comment by Biswa96** [comment]
> As shown in the second picture, it's a workaround for mingw.

The workaround is incorrect in this context. I have found appropriate solutions for those errors. I would like to create a separate PR after merging this one. Does that sound good to you?

**Comment by ArchGuyWu** [comment]
> > As shown in the second picture, it's a workaround for mingw.
> 
> The workaround is incorrect in this context. I have found appropriate solutions for those errors. I would like to create a separate PR after merging this one. Does that sound good to you?

You can directly modify my fork.

**Comment by ArchGuyWu** [comment]
But it's ok if you want to make another PR after merging this.

**Pull request merged**